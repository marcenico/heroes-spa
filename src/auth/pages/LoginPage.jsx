import React, { useContext } from 'react';
import { useNavigate } from 'react-router-dom';
import { AuthContext } from '../context';

export const LoginPage = () => {
  const navigate = useNavigate();
  const { login } = useContext(AuthContext);

  const onLogin = () => {
    const lastPath = localStorage.getItem('lastPath') || '/';
    login('Marcelo');
    navigate(lastPath, { replace: true });
  };

  return (
    <div className="container mt-5">
      <h1>LoginPage</h1>
      <hr />

      <button className="btn btn-primary" type="button" onClick={onLogin}>
        Login
      </button>
    </div>
  );
};
