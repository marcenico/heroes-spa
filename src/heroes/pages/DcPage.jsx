import React from 'react';
import { HeroList } from '../components';

export const DcPage = () => {
  return (
    <>
      <h1>DC</h1>
      <hr />

      <HeroList publisher="DC Comics" />
    </>
  );
};
