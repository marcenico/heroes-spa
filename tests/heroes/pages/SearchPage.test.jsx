import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from '../../../src/heroes/pages/SearchPage';

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate
}));

describe('Pruebas en <SearchPage />', () => {
  beforeEach(() => jest.clearAllMocks());

  test('debe de mostrar el componente correctamente con valores por defecto', () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar a batman y el input para buscar a batman', () => {
    const queryValue = 'batman';

    render(
      <MemoryRouter initialEntries={[`/search?q=${queryValue}`]}>
        <SearchPage />
      </MemoryRouter>
    );

    const input = screen.getByRole('textbox');
    const img = screen.getByRole('img');
    const alertDiv = screen.getByTestId('alert-danger');

    expect(input.value).toBe(queryValue);
    expect(img.src).toContain('/assets/heroes/dc-batman.jpg');
    expect(alertDiv.style.display).toBe('none');
  });

  test('debe de mostrar un error si no se encuentra el hero (batman123)', () => {
    const queryValue = 'batman123';

    render(
      <MemoryRouter initialEntries={[`/search?q=${queryValue}`]}>
        <SearchPage />
      </MemoryRouter>
    );

    const alertDiv = screen.getByTestId('alert-danger');

    expect(alertDiv.style.display).toBe('');
  });

  test('debe de llamar el navigate a la pagina nueva', () => {
    const inputValue = 'superman';

    render(
      <MemoryRouter initialEntries={['/search']}>
        <SearchPage />
      </MemoryRouter>
    );

    const input = screen.getByRole('textbox');
    const form = screen.getByTestId('form');

    fireEvent.change(input, { target: { name: 'searchText', value: inputValue } });
    fireEvent.submit(form);

    expect(mockedUseNavigate).toHaveBeenCalledWith(`?q=${inputValue}`);
  });
});
