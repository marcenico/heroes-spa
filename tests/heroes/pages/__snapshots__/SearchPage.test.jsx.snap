// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> debe de mostrar el componente correctamente con valores por defecto 1`] = `
<div>
  <h1>
    SearchPage
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
      <div
        class="alert alert-danger animate__animated animate__fadeIn"
        data-testid="alert-danger"
        style="display: none;"
      >
        No hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
